packages/core-modules/contracts/modules/ElectionModule.sol

```diff
@@ -1,7 +1,133 @@
+
+        if (store.currentEpochIndex != 0) {
+            revert InitError.AlreadyInitialized();
+        }
+
+        store.currentEpochIndex = 1;
+
+        _configureFirstEpoch(epochEndDate, nominationPeriodStartDate, votingPeriodStartDate);
+    }
+
+    function adjustEpoch(
+        uint64 epochEndDate,
+        uint64 nominationPeriodStartDate,
+        uint64 votingPeriodStartDate
+    ) external override onlyOwner onlyWithStatus(EpochStatus.Idle) {
+        EpochData storage epoch = _getCurrentEpoch();
+
+        // TODO: Validate that the new dates are in the future and/or are close to the initial dates?
+
+        _configureEpoch(epoch, epoch.startDate, epochEndDate, nominationPeriodStartDate, votingPeriodStartDate);
+    }
+
+    /* solhint-disable */
+    function nominate() external override onlyWithStatus(EpochStatus.Nominating) {
+        // TODO
+    }
+
+    /* solhint-enable */
+
+    /* solhint-disable */
+    function withdrawNomination() external override onlyWithStatus(EpochStatus.Nominating) {
+        // TODO
+    }
+
+    /* solhint-enable */
+
+    /* solhint-disable */
+    function elect(address[] memory candidates) external override onlyWithStatus(EpochStatus.Voting) {
+        // TODO
+    }
+
+    /* solhint-enable */
+
+    function evaluate() external override onlyWithStatus(EpochStatus.Evaluating) {
+        // TODO
+
+        _getCurrentEpoch().evaluated = true;
+    }
+
+    function resolve() external override onlyWithStatus(EpochStatus.Evaluating) {
+        if (!isCurrentEpochEvaluated()) {
+            revert EpochNotEvaluated();
+        }
+
+        // TODO: Shuffle NFTs
+
+        _getCurrentEpoch().resolved = true;
+
+        _configureNextEpoch();
+
+        ElectionStore storage store = _electionStore();
+        store.currentEpochIndex = store.currentEpochIndex + 1;
+    }
+
+    // ----------------------------------
+    // Current epoch views
+    // ----------------------------------
+
+    function getEpochStatus() public view override returns (uint) {
+        return uint(_getEpochStatus());
+    }
+
+    function getEpochIndex() public view override returns (uint) {
+        return _electionStore().currentEpochIndex;
+    }
+
+    function getEpochStartDate() public view override returns (uint64) {
+        return _getCurrentEpoch().startDate;
+    }
+
+    function getEpochEndDate() public view override returns (uint64) {
+        return _getCurrentEpoch().endDate;
+    }
+
+    function getNominationPeriodStartDate() public view override returns (uint64) {
+        return _getCurrentEpoch().nominationPeriodStartDate;
+    }
+
+    function getVotingPeriodStartDate() public view override returns (uint64) {
+        return _getCurrentEpoch().votingPeriodStartDate;
+    }
+
+    function isCurrentEpochEvaluated() public view override returns (bool) {
+        return _getCurrentEpoch().evaluated;
+    }
+
+    // ----------------------------------
```

Comment:
What happens with all the nextEpoch* when there's no no next epoch (during initialization period)?

Reply:
Before initialization:
My plan is to use the onlyIfInitialized modifier that you're working on.

After initialization:
There should always be a a next epoch.

Reply:
Actually, you're right! The next epoch will be retrievable as per my previous message, but it will be empty. Removing all getters for nextEpoch* for now.